crates/re_space_view_spatial/src/visualizers/transform3d_arrows.rs

```diff
@@ -50,11 +49,21 @@ impl VisualizerSystem for Transform3DArrowsVisualizer {
         query: &ViewQuery<'_>,
         view_ctx: &ViewContextCollection,
     ) -> Result<Vec<re_renderer::QueueableDrawData>, SpaceViewSystemExecutionError> {
-        let mut line_builder = view_ctx.get::<SharedRenderBuilders>()?.lines();
         let transforms = view_ctx.get::<TransformContext>()?;
 
         let store = ctx.entity_db.store();
         let latest_at_query = re_data_store::LatestAtQuery::new(query.timeline, query.latest_at);
+
+        // Counting all transform ahead of time is a bit wasteful and we don't expect a huge amount of lines from them,
+        // so use the `LineDrawableBuilderAllocator` utility!
```

Comment:
This makes it sound like `LineDrawableBuilderAllocator` support dynamic growing?

If so, why do all the math below?

Reply:
I mean.. yeah, kinda? It gives you new line builders once one is full